### PR TITLE
Miscellaneous Bug Fixes

### PR DESCRIPTION
**Pull request by DamianMadden**
Ignoring whitespace changes for this eliminates most of the changes, alot just gets wrapped in a while loop

### CODE REVIEW

File: src/vcBPA.cpp [hunk 1]
```diff
@@ -387,9 +387,6 @@ vcBPATriangle vcBPA_FindSeedTriangle(vcBPAGrid *pGrid, double ballRadius)
         {
           triangle.x = (uint64_t)nearbyPoints[i];
           triangle.y = (uint64_t)pointIndex;
-          a = p0 - p1;
-          b = p2 - p1;
-          triangleNormal = udNormalize(udCross(a, b));
```
Comment (samsinsane) [inline src/vcBPA.cpp]: This isn't a bug fix?

File: src/vcBPA.cpp [hunk 2]
```diff
@@ -923,87 +920,94 @@ vdkError vcBPA_ConvertReadPoints(vdkConvertCustomItem *pConvertInput, vdkPointBu
   // Reset point count to avoid infinite loop
   pBuffer->pointCount = 0;
 
-  vcBPAConvertItem *pData = (vcBPAConvertItem *)pConvertInput->pData;
-  bool getNextGrid = true;
+  while (pBuffer->pointCount == 0)
```
Comment (samsinsane) [inline src/vcBPA.cpp]: What bug does this fix?

File: src/vcBPA.cpp [hunk 3]
```diff
@@ -923,87 +920,94 @@ vdkError vcBPA_ConvertReadPoints(vdkConvertCustomItem *pConvertInput, vdkPointBu
...
 
-    udDouble3 position = pData->activeItem.pGrid->vertices[i].position;
-    size_t triangle = vcBPA_FindClosestTriangle(&pData->activeItem.oldGrid, position);
+      udDouble3 position = pData->activeItem.pGrid->vertices[i].position;
 
-    double distance = FLT_MAX;
-    if (triangle < pData->activeItem.oldGrid.triangles.length)
-      distance = udAbs(vcBPA_DistanceToTriangle(&pData->activeItem.oldGrid, triangle, position));
+      if (!udPointInAABB(position, pData->activeItem.pGrid->minPos, pData->activeItem.pGrid->maxPos))
+          continue;
```
Comment (samsinsane) [inline src/vcBPA.cpp]: What bug does this fix?

Comment (DamianMadden) [reply src/vcBPA.cpp]: multiple inclusion of points that were outside the AABB for the sake of the algorithm to complete boundary triangles

File: src/vcBPA.cpp (hunk 2 above)
Comment (DamianMadden) [reply src/vcBPA.cpp]: with the exclusion of points outside the AABB of a grid it's possible that grids are present with 0 points, this prevents early termination of the conversion by proceeding to subsequent grids

File: src/vcBPA.cpp (hunk 1 above)
Comment (DamianMadden) [reply src/vcBPA.cpp]: Not precisely...

File: src/vcBPA.cpp (hunk 3 above)
Comment (samsinsane) [reply src/vcBPA.cpp]: `pGrid` shouldn't contain any points outside the AABB - we don't mesh `pGrid`, we mesh `pOldGrid`. If `pGrid` contains any points outside the AABB, that should be fixed instead.

**Comment by OsiJr** [comment]
Replaced by #291

**Pull request closed without merging**